src/test/ui/consts/duration-consts-2.rs

```diff
@@ -0,0 +1,58 @@
+// run-pass
+
+#![feature(const_panic)]
+#![feature(const_if_match)]
+#![feature(duration_consts_2)]
+#![feature(div_duration)]
+
+use std::time::Duration;
+
+fn duration() {
+    const ZERO : Duration = Duration::new(0, 0);
+    assert_eq!(ZERO, Duration::from_secs(0));
+
+    const ONE : Duration = Duration::new(0, 1);
+    assert_eq!(ONE, Duration::from_nanos(1));
+
+    const MAX : Duration = Duration::new(u64::MAX, 1_000_000_000 - 1);
+
+    const MAX_ADD_ZERO : Option<Duration> = MAX.checked_add(ZERO);
+    assert_eq!(MAX_ADD_ZERO, Some(MAX));
+
+    const MAX_ADD_ONE : Option<Duration> = MAX.checked_add(ONE);
+    assert_eq!(MAX_ADD_ONE, None);
+
+    const ONE_SUB_ONE : Option<Duration> = ONE.checked_sub(ONE);
+    assert_eq!(ONE_SUB_ONE, Some(ZERO));
+
+    const ZERO_SUB_ONE : Option<Duration> = ZERO.checked_sub(ONE);
+    assert_eq!(ZERO_SUB_ONE, None);
+
+    const ONE_MUL_ONE : Option<Duration> = ONE.checked_mul(1);
+    assert_eq!(ONE_MUL_ONE, Some(ONE));
+
+    const MAX_MUL_TWO : Option<Duration> = MAX.checked_mul(2);
+    assert_eq!(MAX_MUL_TWO, None);
+
+    const ONE_DIV_ONE : Option<Duration> = ONE.checked_div(1);
+    assert_eq!(ONE_DIV_ONE, Some(ONE));
+
+    const ONE_DIV_ZERO : Option<Duration> = ONE.checked_div(0);
+    assert_eq!(ONE_DIV_ZERO, None);
+
+    const MAX_AS_F32 : f32 = MAX.as_secs_f32();
+    assert_eq!(MAX_AS_F32, u64::MAX as f32 + 0.000_000_000_1);
```

Comment:
Why `0.000_000_000_1`?

Reply:
that's what clippy suggests as the format for float literals

Reply:
or are you asking why not `u64::EPSILON`?

Reply:
I was asking why this specific value. It seems the max number of nanosw as added so shouldn't it be `0.999_999_999`?
No idea what `u64::EPSILON` is.

Reply:
[`f64::EPSILON`](https://doc.rust-lang.org/std/primitive.f64.html#associatedconstant.EPSILON) is the smallest value you can add to `1.0` that will change the float.

I'm assuming that the `0.000_000_000_1` here is supposed to attempt the same: add a very small value to the result of this conversion and check whether that addition changes anything.

Reply:
Ah, `f64::EPSILON` makes more sense than `u64::EPSIOLON`. :)

Reply:
oh darn. I didn't notice that typo at all